pkg/apis/storage/validation/validation.go

```diff
@@ -164,8 +164,37 @@ func validateAttacher(attacher string, fldPath *field.Path) field.ErrorList {
 // validateSource tests if the source is valid for VolumeAttachment.
 func validateVolumeAttachmentSource(source *storage.VolumeAttachmentSource, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
-	if source.PersistentVolumeName == nil || len(*source.PersistentVolumeName) == 0 {
```

Comment:
this field was previously marked as optional, but required by validation. that means an n-1 apiserver would see a persisted object with `inlineVolumeSource` specified as invalid (it wouldn't know about `inlineVolumeSource` and would think the volumeAttachment had not specified any source)

need to think about how to make the addition of this field safe for rollback

Reply:
@liggitt in the case of rollback, where `inlineVolumeSource`  is present, but not `persistentVolumeName`, the validation would fail.  I am wrapping the validation in a feature-gate condition forcing an explicit enabling of this feature.

Reply:
Strictly speaking there's no way to convert a field from required to optional and be rollback-safe.  This is kind of a problem.  Any creative ideas here?